pkg/operator/ceph/object/notification/provisioner.go

```diff
@@ -0,0 +1,207 @@
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package notification to manage a rook bucket notifications.
+package notification
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	awssession "github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/s3"
+	"github.com/ceph/go-ceph/rgw/admin"
+	"github.com/coreos/pkg/capnslog"
+	bktv1alpha1 "github.com/kube-object-storage/lib-bucket-provisioner/pkg/apis/objectbucket.io/v1alpha1"
+	"github.com/pkg/errors"
+	"github.com/rook/rook/pkg/apis/ceph.rook.io/s3ext"
+	cephv1 "github.com/rook/rook/pkg/apis/ceph.rook.io/v1"
+	"github.com/rook/rook/pkg/clusterd"
+	cephclient "github.com/rook/rook/pkg/daemon/ceph/client"
+	"github.com/rook/rook/pkg/operator/ceph/object"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type Provisioner struct {
+	Client      client.Client
+	Context     *clusterd.Context
+	ClusterInfo *cephclient.ClusterInfo
+	ClusterSpec *cephv1.ClusterSpec
+	sess        *awssession.Session
+}
+
+func (p *Provisioner) HasSession() bool {
+	return p.sess != nil
+}
+
+func (p *Provisioner) getCephUser(username string, objStore *cephv1.CephObjectStore, objContext *object.Context) (accessKey string, secretKey string, err error) {
+	if len(username) == 0 {
+		err = errors.New("no user name provided")
+		return
+	}
+
+	// CephClusterSpec is needed for GetAdminOPSUserCredentials()
+	objContext.CephClusterSpec = *p.ClusterSpec
+	adminAccessKey, adminSecretKey, err := object.GetAdminOPSUserCredentials(objContext, &objStore.Spec)
+	if err != nil {
+		return
+	}
+
+	adminOpsClient, err := admin.New(objContext.Endpoint, adminAccessKey, adminSecretKey, &http.Client{})
+	if err != nil {
+		err = errors.Wrapf(err, "failed to build admin ops API connection to get user: %q", username)
+		return
+	}
+
+	var u admin.User
+	u, err = adminOpsClient.GetUser(context.TODO(), admin.User{ID: username})
+	if err != nil {
+		err = errors.Wrapf(err, "failed to get ceph user: %q", username)
+		return
+	}
+
+	logger.Infof("successfully fetched ceph user: %q", username)
+	accessKey = u.Keys[0].AccessKey
+	secretKey = u.Keys[0].SecretKey
+	return
+}
+
+func (p *Provisioner) CreateSession(owner string, objectStoreName types.NamespacedName) error {
+	p.sess = nil
+	objStore, err := p.Context.RookClientset.CephV1().CephObjectStores(objectStoreName.Namespace).Get(context.TODO(), objectStoreName.Name, metav1.GetOptions{})
+	if err != nil {
+		return err
```

Comment:
I generally find that calls to the k8s api don't always return the most clear information about what is wrong. I would suggest adding even simple context here like `failed to get CephObjectStore <namespacedName>`.